pkg/controller/sync/aggregator/aggregator.go

```diff
@@ -0,0 +1,105 @@
+package aggregator
+
+import (
+	"fmt"
+	gosync "sync"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type KindName struct {
```

Comment:
"Kind" is used in two different contexts in this package (the "kind" of the GVK to sync vs. the "kind" which identifies the kind that's sourcing the config). This will get confusing.

Maybe something like:

```
type Key struct {
   Source string
   ID string
}
```

where "source" identifies the source of the intent (same as "kind", but this also leaves room for non-k8s-based sources, like command line flags), and ID is essentially name.

I'm open to other ideas, but we should try to avoid overloading "kind"

Reply:
Reworded